### DISCUSSION TITLE
add some kind of public sharing option for generated sites?

### DISCUSSION THREAD

**Discussion by leonstafford** [elementor/lokl#42] [Ideas]
not a fan of anything that open more security holes, but as the user has complete control over customizing their environment, giving an easier way to opt-in to a common solution may be useful.

examples of options:

 - ngrok
 - Expose
 
My preferred workflow would be to simply publish the static site to a preview URL vs a live site (ie, preview option if using Netlify's CLI client). But, this may be something people want

**Comment by shinyface1111** [comment]
Sorry I just discovered your question in this discussion area.  Leon, I absolutely am amazed by what you've spent time creating here.  I'm a beginner so really just touching on understanding what you've accomplished with LOKL containerization.  I know it just works and it's easy to find.  

In my case, even on v 5.0.0, I'm getting the dreaded 502 bad gateway error (nginx 1.18.0), whenever I log out of my mac OS Big Sur ,v 11.6 so sites I've spun up aren't accessible. I have confidence they are still here though, so I'm on the hunt right now for a "rough" video you mentioned that you created re Docker.  

BTW, I'm an early adopter for the Brizy.io web builder and I've shared with the creators and users on the user forum that LOKL has been ideal for learning in a secure protected localhost environment.  

Thanks for the amazing intellectual puzzles you've spent the past couple of years solving. I wish there was a way to help you sustain all this... as you've talked about.  The problems you have addressed have aided the rest of us in focusing on what we want to share, without having to go through years of learning, just to plateau to where you were as a gradeschooler.  Thank you Leon

**Comment by leonstafford** [comment]
Hi @shinyface1111,

Thanks for the kind and motivating words, really appreciate it!

I prepared a new Lokl build this week, with latest versions of all software and some new ones.

Would you be available some time this week to do a screenshare, so I can help you get back into your Lokl instances and debug what the issue is?

If you create a brand new Lokl site using 5.0.0, does this issue still occur or is it only with ones created with an earlier version?

Looking forward to help sort this for you!

Cheers,

Leon

Can email me me@ljs.dev if suitable/OK to screenshare, else we can try n work through it asynchronously

**Comment by shinyface1111** [comment]
Hey Leon,
I really wasn't expecting you to get back to me. THANK YOU!  I'd love to take you up on that. Reading some help you've offered to other people and a lot of Docker help, I've dumped Docker, reinstalled and did the same for LOKL.  In the past 12 hours, I reinstalled and I'm being very diligent about logging out and shutting everything down, then on reboot, I'm not getting the 502s - everything is working as you designed so far. Not sure I've been able to maintain some of the containers, but I'm going to try and find them this afternoon.

Not sure I mentioned that I've shared LOKL w/ people who are using a newer web builder called Brizy.io developed by some designers at Themefuse (https://www.facebook.com/groups/brizy).  They've done something interesting by "building" a product called Brizy Cloud that bypasses WP. It's pretty slick in that it's based on static (your Bailiwick). I mention it because while great, there's no ability to add necessary APIs or plugins so many of us early adopter folks end up going back to WP and then don't understand how to get back to the speed and security, along with lower expense of static WP hosting.  They've also teamed up w/ Netlify, Render, Vercel and recently AWS & Bunny CDN etc, on the cloud side, but it's not so easy to comprehend how to integrate static on WP. 

Might be an opportunity for you guys at Strattic!  I feel like the pricing is a little too high since the point is to use inexpensive hosting.  Maybe they could help promote Strattic and bring pricing down for their users.  As a beginning wordpress guy, who's paid for multiple urls and hosting over the past 5 years and pretty much never published anything live, (there are a lot of us) I feel like the pricing is a little too high since the point is to use inexpensive hosting.  Maybe they could help promote Strattic and bring pricing down for their users. I don't know, I'm in sales so always thinking in that direction.  The creator Dimitrie Baitanciuc, is at: hello@brizy.io and he's really responsive.  So before you up your pricing (which, BTW, I cannot even find any pricing since 4.0) you may want to rethink LOKLs pricing model... of course after I buy it this weekend.  

C'mon, you can't continue to change the world without compensation as generous and altruistic as your mission is.  I mean seriously? You're coding, making changes, working with the Strattic website folks and offering to help the random beginner who gets stuck using your coding, by offering to do a screenshare! We've got to come together to keep you healthy - you're changing the world.  Thank you Leon. I'll let you know how things go this weekend before taking up any of your time.  In anticipation of success finding what I've created, thank you so much, your work has given me a greater level of confidence at getting my message out.

Best

Dave

**Reply by leonstafford** [reply]
Thanks, not least for expanding my vocabulary with _Bailiwick_!

Lokl itself is free and I don't envision ever charging for Lokl (also, I like you capitalising it, I think that helps people from misreading it as Lokl, with all the superhero movies around these days! I'll move to all caps when finally branding it!

The plugins within, like WP2Static, are all freely available as source code on GitHub, but I do try to drive sales for easy to install zip files on wp2static.com

We're (partnered up with 2 great devs who've been contributing to WP2Static's code for the past year) taking things in a more commercial direction soon, but still keeping the code fully open source. I'll never have qualms about offering discounted/free licenses to any users who request them (like our Fair Go Pricing concept: https://wp2static.com/blog/fair-go-pricing/).

So, if you want to purchase WP2Static, now's a good time, as it will lock you into a lifetime, unlimited site license for $100, whereas in future, it will likely be recurring subscriptions, maybe site limited, too. But, you also don't need to do that anytime soon.

Thanks for the interesting info re Brizy Cloud, will think on it and also share with the Strattic team. I'd heard good things about Brizy since its inception and believe it's always played nicely with static site conversion.

**Reply by leonstafford** [reply]
I have a feeling if there's other Lokl sites that aren't spinning up for you, that they were created with an older version, but I'll be able to help you get those launching again

**Comment by shinyface1111** [comment]
Hey Leon,

Long time no talk to.  I never followed up regarding your offer to do a
screenshare about the problem with getting access to websites I couldn't
access using Lokl 4.0.  Didn't want to take up your "free" time since you'd
just signed on to your new project.  Redoing the sites was a good exercise
in developing workflow.

Thanks for the tip though! I did purchase WP2Static prior to your merging
into Strattic.  No need to busy myself (yet) with the migration process.
By the way, WP2static is working out wonderfully and your documentation for
a non-coder newbie has walked me through the steps that have provided a
free, fast, secure, static, no worries web experience.  Most importantly,
the basic dynamic stuff I've been able to incorporate all works great!
Followed your documentation and separated out the ecommerce stuff through
Ecwid although I read you have the most success with and recommend
Snipcart.  I'll move over to that soon.

Just wanted to share with you that I recently started working on a site
with an all-in-one theme by WP Estate, based on Airbnb booking and they're
using Visual Studio as for their localhost development platform. Here's one
of the demos on youtube - https://www.youtube.com/watch?v=KrIWmhgSKqA

The theme plugin requirements are extremely heavy. I don't know if the lag
is on my macbook, which is a 2015 macbook pro 2.7 GHz Dual-Core Intel Core
i5, with 8 GB 1867 MHz DDR3 or if it's a problem with too many plugins
bogging down Lokl.  I've reinstalled all the plugins numerous times and
made new WP installs a few times and switched from Safari to Chrome and
finally to Firefox and now the localhost delay is bearable.  On Safari, I'd
wait upwards of 8 minutes after a click in order for the prompt to take
effect and even then I'd run into a lot of errors.  Firefox is still slow
compared to other development sites I've created with Lokl.  Maybe just too
resource heavy on my computer?  I know the plugins are most likely the
problem, but unfortunately they are required. My alternative is to
subscribe to this identical booking engine functionality for the AirBnb at
$89-$400/month plus $3000 setup and content insertion.  BTW this theme cost
$60!  I think we're in the wrong business!

Here's the list of required plugins:

   1. WP Rentals Theme Core Functionality Build
   2. WP Bakery (crazy that you have to have both theme builders installed
   even though technically I will not use any WP Bakery functionality*)
   3. Elementor (" " " *)
   4. Exclusive Web GL
   5. Slider Revolution
   6. Ultimate Addons for WP Bakery*
   7. WP Rentals Elementor Widgets
   8. WP Rentals Gutenberg Blocks

I don't want nor expect you to spend any time on this. Just wondered if,
off the top of your head, you had any experience with this level of plugin
bloat or had any suggestions, since I watch them build on localhost and
even with editing, whatever they use seems to work decently. Alex, one of
the developers mentions using Visual Studio at 1:35
https://www.youtube.com/watch?v=KrIWmhgSKqA

Leon, I hope you are well and healthy. Thank you again so much for opening
up so many opportunities and making them understandable for the many people
across the world that you have helped.  Best ahead for you and the
development team at Strattic. Some of your colleagues are probably amidst
the conflict areas and my wishes for theirs and their family's safety is
more important than anything.

Best Regards and have a nice weekend.

Dave
480.269.2688
